sstables/sstables.cc

```diff
@@ -1563,7 +1563,8 @@ future<> sstable::load_metadata(sstable_open_config cfg, bool validate) noexcept
     co_await coroutine::all(
             [&] { return read_compression(); },
             [&] { return read_filter(cfg); },
-            [&] { return read_summary(); });
+            [&] { return read_summary(); },
+            [&] { return read_checksum(); });
     if (validate) {
```

Comment:
@lkshminarayanan do we need integration into sstables_manager for accounting? output of `nodetool tablestats`?

Benchmark: if everything else is already accounted in sstables_manager, this need to be accounted too, otherwise we can defer it to a later patch.

Reply:
> @lkshminarayanan do we need integration into sstables_manager for accounting?

I don't think this is needed for now, since this is only used in scrub, we can load before and release after use. Once we start using it more widely, we will need to reconsider caching it.

> output of nodetool tablestats?

How is this related? Adding the memory taken by checksum to some memory stat?

Reply:
Yes, IIRC is shows the bloom filter there.

Not sure if it has a separate entry for other things.

Reply:
I looked at the column_family metrics API. It reports the memory used by the memtable, bloom filter, index summary and compression metadata. There is nothing about CRC. This is aligned with C*.

I guess we can extend it with a new endpoint for CRC as well. This will be returning zero for compressed SSTables or uncompressed SSTables for which the component is not currently loaded. As @denesb mentioned, there is no need to cache checksums at this point, so even for uncompressed SStables this will be returning zero most of the time.

https://opensource.docs.scylladb.com/stable/operating-scylla/nodetool-commands/tablestats.html
https://docs.datastax.com/en/cassandra-oss/3.x/cassandra/tools/toolsTablestats.html

Reply:
I think we can defer this to when we start keeping `checksum` components in memory. I don't think it is interesting to see 0 there all the time, unless there is a scrub going on.